### PR TITLE
Test case for internal/tls package

### PR DESCRIPTION
**Pull request by kevindiu**

### CODE REVIEW

**Comment by kpango** [comment]
Overall looks good, just one point.
could you please implement TestNewClientConfig ?

**Comment by kevindiu** [comment]
I can't push to this branch...
```
MBP13UAQ-A070:tls wdiu$ git push
Username for 'https://github.com': kevindiu
Password for 'https://kevindiu@github.com':
Enumerating objects: 13, done.
Counting objects: 100% (13/13), done.
Delta compression using up to 4 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (8/8), 3.05 KiB | 3.05 MiB/s, done.
Total 8 (delta 5), reused 0 (delta 0)
remote: Resolving deltas: 100% (5/5), completed with 5 local objects.
remote: error: GH006: Protected branch update failed for refs/heads/test_tls.
remote: error: Required status check "license/cla" is expected.
To https://github.com/vdaas/vald
 ! [remote rejected] test_tls -> test_tls (protected branch hook declined)
error: failed to push some refs to 'https://github.com/vdaas/vald'
```

File: internal/tls/option.go [hunk 1]
```diff
@@ -22,48 +22,49 @@ import "crypto/tls"
...
+		// tls.TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, // Maybe this is work on TLS 1.2
+		// tls.TLS_RSA_WITH_3DES_EDE_CBC_SHA, // TLS1.3 Feature
+		// tls.TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA, // TLS1.3 Feature
+		// tls.TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305, // Go 1.8 only
+		// tls.TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305, // Go 1.8 only
+		// },
+		ClientAuth: tls.NoClientCert,
+	}
 	defaultOpts = []Option{
-		WithTLSConfig(&tls.Config{
```
Comment (kpango) [inline internal/tls/option.go]: inlineing is better

Comment (kevindiu) [reply internal/tls/option.go]: I use it in the test case

Comment (kpango) [reply internal/tls/option.go]: I know, but you shouldn't use field variable for FOP testing.
you can overwrite configuration by FOP in testing

File: internal/tls/tls_test.go [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+				}
+				return nil
+			},
+		},
+		{
+			name: "return value CurvePreferences test.",
+			args: defaultArgs,
+			want: func() *Config {
+				conf := defaultConfig
+				cert, _ := tls.LoadX509KeyPair(dummyCertPath, dummyKeyPath)
```
Comment (kpango) [inline internal/tls/tls_test.go]: please check error anytime

**Comment by vankichi** [comment]
It is no longer updated.
We should do this task as a new PR.

**Pull request closed without merging**